### PR TITLE
feat: add marketing opt-in

### PR DESCRIPTION
**Pull request by jimsynz**
Adds the ability for new users to opt in to marketing communications when they sign up and to change their opt-in status on their settings page.

### CODE REVIEW

File: lib/playdate_web/templates/user_registration/new.html.eex [hunk 1]
```diff
@@ -18,6 +18,17 @@
     <%= error_tag f, :password %>
   </div>
 
+  <div class="mb-4">
+    <%= label f, :marketing_opt_in, class: "form-label" %>
+    <%= checkbox f, :marketing_opt_in, class: "form-checkbox" %>
+
+    <p>
+      <a href="https://alembic.com.au/">Alembic</a> collects user's email
```
Comment (juddey) [inline lib/playdate_web/templates/user_registration/new.html.eex:26]:
```suggestion
      <a href="https://alembic.com.au/">Alembic</a> collects your email
```

File: lib/playdate_web/templates/user_registration/new.html.eex [hunk 2]
```diff
@@ -18,6 +18,17 @@
...
+      addresses for the sole purpose of marketing our products and services.
```
Comment (juddey) [inline lib/playdate_web/templates/user_registration/new.html.eex:27]:
```suggestion
      address for the sole purpose of marketing our products and services.
```

File: lib/playdate_web/templates/user_registration/new.html.eex [hunk 3]
```diff
@@ -18,6 +18,17 @@
...
+      No user data will be shared with third parties.
```
Comment (juddey) [inline lib/playdate_web/templates/user_registration/new.html.eex:28]:
```suggestion
      Your data will not be shared with third parties.
```

**Review by Theosaurus-Rex — APPROVED** [review]
Works great, nicely done!

File: lib/playdate_web/templates/user_settings/edit.html.eex [hunk 4]
```diff
@@ -68,6 +68,27 @@
           </div>
         <% end %>
 
+        <h4>Marketing opt <%= if @current_user.marketing_opt_in, do: "out", else: "in" %></h4>
+
+        <p>
+          <a href="https://alembic.com.au/">Alembic</a> collects user's email
```
Comment (juddey) [inline lib/playdate_web/templates/user_settings/edit.html.eex:74]:
```suggestion
          <a href="https://alembic.com.au/">Alembic</a> collects your email
```

File: lib/playdate_web/templates/user_settings/edit.html.eex [hunk 5]
```diff
@@ -68,6 +68,27 @@
...
+          addresses for the sole purpose of marketing our products and services.
```
Comment (juddey) [inline lib/playdate_web/templates/user_settings/edit.html.eex:75]:
```suggestion
          address for the sole purpose of marketing our products and services.
```

File: lib/playdate_web/templates/user_settings/edit.html.eex [hunk 6]
```diff
@@ -68,6 +68,27 @@
...
+          No user data will be shared with third parties.
```
Comment (juddey) [inline lib/playdate_web/templates/user_settings/edit.html.eex:76]:
```suggestion
          Your data will not be shared with third parties.
```

**Comment by juddey** [comment]
Added a few wording updates. Might refactor that note in case we need to use it in other places, but not needed for merge @jimsynz :+1:

**Review by juddey — APPROVED** [review]

**Comment by jimsynz** [comment]
Thanks @juddey - I'm going to wait for @Theosaurus-Rex's PR to land and rebase on top of it. I'll address these suggestions then.

**Pull request merged**